src/simpleMask.js

```diff
@@ -0,0 +1,159 @@
+        if(selectedCharacterPattern.length !== 1 || !requiredParamsArePresent) {
+            settingsAreValid = false;
+        }
+
+        return settingsAreValid;
+    }
+
+    const initPatternData = () => {
+        const formatArrayOfCharacters = settings.format.split('');
+        arrayOfSpecialCharactersInFormat = getSpecialCharacters(formatArrayOfCharacters);
+    }
+
+    const getSpecialCharacters = arrayOfCharacters => {
+        let specialCharactersArray = [];
+
+        arrayOfCharacters.forEach((char, index) => {
+            if (char !== settings.maskPlaceholder) {
+                specialCharactersArray.push({ char, index });
+            }
+        });
+
+        return specialCharactersArray;
+    };
+
+    const bindEvents = () => {
+        inputEl.addEventListener('focusin', maskInput);
+        inputEl.addEventListener('focusout', maskInput);
+        inputEl.addEventListener('keypress', checkForInvalidKey);
+    };
+
+    const maskInput = (e) => {
+        if (inputEl.value.length !== 0 && inputEl.value.length >= getFormatLength() ) {
+            const value = getMaskValue(arrayOfSpecialCharactersInFormat, getMaskMethod(e));
+            setInputValue(value);
+
+            if (settings.onMaskCallback) {
+                settings.onMaskCallback(value)
+            }
+        }
+    };
+
+    const getMaskMethod = (e) => e.type === 'focusin' ? maskMethods.reverse : maskMethods.forward;
+
+    const setInputValue = (value) => inputEl.value = value;
+
+    const getMaskValue = (arrayOfSpecialCharacters, maskMethod) => {
+        return arrayOfSpecialCharacters.reduce((maskedValue, specialCharObject) => {
+            maskedValue = buildMaskedValue(specialCharObject, maskMethod, maskedValue);
+            return maskedValue;
+        }, '');
+    };
+
+    const buildMaskedValue = (specialCharObject, maskMethod, maskedValue) => {
+        const valueToBuildFrom =  maskedValue.length === 0 ? inputEl.value : maskedValue;
+
+        if(maskMethod === maskMethods.forward ) {
+            output = valueToBuildFrom.substring(specialCharObject.index, specialCharObject.char) + specialCharObject.char + valueToBuildFrom.substring(specialCharObject.index);
+        } else {
+            output = valueToBuildFrom.split(specialCharObject.char).join("")
+        }
+
+        return output;
+    };
+
+
+    const checkForInvalidKey = e => {
+        if (inputEl.value.length === getFormatLength() || !keyMatchesPattern(e.key)) {
+            e.preventDefault();
+
+            if(settings.onFailedInputCallback) {
+                settings.onFailedInputCallback(inputEl.value);
+            }
+        }
+    };
+
+    const getFormatLength = () => {
+        let formatLength;
+
+        if (arrayOfSpecialCharactersInFormat.length !== 0) {
+            formatLength = settings.format.length - arrayOfSpecialCharactersInFormat.length;
+        } else {
+            formatLength = settings.format.length;
+        }
+
+        return formatLength;
```

Comment:
[suggestion] I think you can turn this entire function body into a one-liner: 
`return settings.format.length - arrayOfSpecialCharactersInFormat.length;`
If the length of `arrayOfSpecialCharactersInFormat` is zero, you'd be subtracting zero, so no harm.